src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor.py

```diff
@@ -79,22 +80,18 @@ def _get_value_from_tuples(
 class AuthenticateServerInterceptor(grpc.ServerInterceptor):  # type: ignore
     """Server interceptor for client authentication."""
 
-    def __init__(
-        self,
-        client_public_keys: Set[bytes],
-        private_key: ec.EllipticCurvePrivateKey,
-        public_key: ec.EllipticCurvePublicKey,
-    ):
-        self.server_private_key = private_key
-        self.client_public_keys = client_public_keys
-        self.encoded_server_public_key = base64.urlsafe_b64encode(
-            public_key_to_bytes(public_key)
-        )
-        log(
-            INFO,
-            "Client authentication enabled with %d known public keys",
-            len(client_public_keys),
-        )
+    def __init__(self, state: State):
+        self.state = state
+        self._lock = threading.Lock()
+        self.server_private_key: Optional[ec.EllipticCurvePrivateKey] = None
+        private_key = self.state.get_server_private_key()
+        public_key = self.state.get_server_public_key()
+        if private_key is not None:
+            self.server_private_key = bytes_to_private_key(private_key)
+        self.client_public_keys = state.get_client_public_keys()
+        self.encoded_server_public_key: Optional[bytes] = None
+        if public_key is not None:
```

Comment:
Same as above: shouldn't we treat this as an unrecoverable error?

Reply:
Yes. Implemented.